### PR TITLE
fix(monitor): set monitors to nil after closing them

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#10035

### CODE REVIEW

**Comment by derekbit** [comment]
@mergify backport v1.8.x

**Comment by derekbit** [comment]
test passed
https://ci.longhorn.io/job/private/job/longhorn-e2e-test/2292/

File: controller/monitor/environment_check_monitor.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func NewEnvironmentCheckMonitor(logger logrus.FieldLogger, ds *datastore.DataSto
 	ctx, quit := context.WithCancel(context.Background())
 
 	m := &EnvironmentCheckMonitor{
-		baseMonitor: newBaseMonitor(ctx, quit, logger, ds, NodeMonitorSyncPeriod),
+		baseMonitor: newBaseMonitor(ctx, quit, logger, ds, DiskMonitorSyncPeriod),
```
Comment (mantissahz) [inline controller/monitor/environment_check_monitor.go:67]: Should the sync period be `EnvironmentCheckMonitorSyncPeriod`?

Comment (derekbit) [reply controller/monitor/environment_check_monitor.go]: Good catch! Thanks.

**Review by mantissahz — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@mergify backport v1.7.x v1.6.x

**Pull request merged**